homeassistant/components/minecraft_server/__init__.py

```diff
@@ -58,11 +66,110 @@ async def async_unload_entry(hass: HomeAssistant, config_entry: ConfigEntry) ->
 
     # Clean up.
     server.stop_periodic_update()
-    hass.data[DOMAIN].pop(unique_id)
+    hass.data[DOMAIN].pop(config_entry_id)
 
     return unload_ok
 
 
+async def async_migrate_entry(hass: HomeAssistant, config_entry: ConfigEntry) -> bool:
+    """Migrate old config entry to a new format."""
+    _LOGGER.debug("Migrating from version %s", config_entry.version)
+
+    # 1 --> 2: Use config entry ID as base for unique IDs.
+    if config_entry.version == 1:
+        assert config_entry.unique_id
+        assert config_entry.entry_id
```

Comment:
The config entry `entry_id` attribute isn't optional.

Reply:
I'll remove the assert of the entry_id in another PR.